src/Variant.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Variant.h"
+
+
+
+bool Variant::toBool()
+{
+    switch (datatype)
+    {
+        case DataTypes::NULL_T:
+            return bool();
+        case DataTypes::BOOL:
+            return b_;
+        case DataTypes::INT:
+            return static_cast<bool>(i_);
+        case DataTypes::FLOAT:
+            return static_cast<bool>(f_);
+        case DataTypes::STRING:
+            return !s_.empty();
+    }
+}
+
+
+int Variant::toInt()
+{
+    switch (datatype)
+    {
+        case DataTypes::NULL_T:
+            return int();
+        case DataTypes::BOOL:
+            return static_cast<int>(b_);
+        case DataTypes::INT:
+            return i_;
+        case DataTypes::FLOAT:
+            return static_cast<int>(f_);
+        case DataTypes::STRING:
+            try
+            {
+                return lexical_cast<int>(s_);
+            }
+            catch (const bad_lexical_cast& blc)
+            {
+                datatype = DataTypes::STRING;
+                return int();
+            }
+    }
+}
+
+
+float Variant::toFloat()
+{
+    switch (datatype)
+    {
+        case DataTypes::NULL_T:
+            return float();
+        case DataTypes::BOOL:
+            return static_cast<float>(b_);
+        case DataTypes::INT:
+            return static_cast<float>(i_);
+        case DataTypes::FLOAT:
+            return f_;
+        case DataTypes::STRING:
+            try
+            {
+                return lexical_cast<float>(s_);
+            }
+            catch (const bad_lexical_cast& blc)
+            {
+                datatype = DataTypes::STRING;
+                return float();
+            }
+    }
+}
+
+
+std::string Variant::toString()
+{
+    switch (datatype)
+    {
+        case DataTypes::NULL_T:
+            return std::string();
+        case DataTypes::BOOL:
+            return lexical_cast<std::string>(b_);
+        case DataTypes::INT:
+            return lexical_cast<std::string>(i_);
+        case DataTypes::FLOAT:
+            return lexical_cast<std::string>(f_);
+        case DataTypes::STRING:
+            return s_;
+    }
+}
+
+
+DataTypes Variant::get_datatype() const
```

Comment:
See contribute file how to define functions

Reply:
> See contribute file how to define functions

Updated